tools/tests/test_ptool.py

```diff
@@ -0,0 +1,45 @@
+# SPDX-License-Identifier: MIT
+# Copyright (c) 2019 Intel Corporation
+# All rights reserved.
+import sys
+import unittest
+import subprocess
+from unittest.mock import patch
+
+from tpm2_pyesys.util.simulator import SimulatorTest
+
+from tpm2_pkcs11 import tpm2_ptool as tool
+from tpm2_pkcs11.utils import TemporaryDirectory
+
+
+class TestPTool(SimulatorTest):
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        cls.abrmd = subprocess.Popen(
+            ["sudo", "-u", "tss", "tpm2-abrmd", "--tcti=mssim"]
```

Comment:
This is super not ideal, I looked at the abrmd repo and am very unclear on what's going on there. Would fixing https://github.com/tpm2-software/tpm2-abrmd/issues/139 make `abrmd` easier to run in a test environment like this?

My code to start the simulator is essentially `tpm_server -rm` followed by connect calls until one of them succeeds then we know it has bound and `SimulatorTest.setUpClass` returns.

Reply:
You can run it in a way where your no on the session bus and then you don't need higher permissions. Look at how we set up the make check tests. We get a simulator and abrmd fired up in this project.